pkg/registry/registry/repository/storage/storage.go

```diff
@@ -174,10 +180,59 @@ func (r *REST) Delete(ctx context.Context, name string, deleteValidation rest.Va
 			return nil, false, err
 		}
 	}
+	repoName := fmt.Sprintf("%s/%s", o.Spec.NamespaceName, o.Spec.Name)
+	repoClient, err := client.NewRepository(repoName, o.Spec.TenantID, r.repoEndpoint, r.tokenPrivateKey)
+	if err != nil {
+		return nil, false, err
+	}
+	tags, err := repoClient.ListTag()
+	if err != nil {
+		return nil, false, err
+	}
+	var wg sync.WaitGroup
+	var errsMap sync.Map
+	for _, tag := range tags {
+		wg.Add(1)
+		go deleteTag(tag, repoClient, errsMap)
+	}
+	err = nil
+	for _, tag := range tags {
+		message, ok := errsMap.Load(tag)
+		if !ok {
+			continue
+		}
+		if err == nil {
+			err = fmt.Errorf("failed to delete %s tag %s: %s", repoClient.Name, tag, message.(string))
+		} else {
+			err = fmt.Errorf("%s, failed to delete %s tag %s: %s", err.Error(), repoClient.Name, tag, message.(string))
+		}
+	}
+	if err != nil {
+		return nil, false, err
+	}
+
```

Comment:
please refine L198-L212 when you check the err

Reply:
done